### PR TITLE
Get top-level block names from .net for .place file

### PR DESCRIPTION
**Pull request by HackerFoo**

### CODE REVIEW

File: common/cmake/devices.cmake [hunk 1]
```diff
@@ -1117,7 +1118,7 @@ function(ADD_FPGA_TARGET)
 
     add_custom_command(
       OUTPUT ${OUT_IO}
-      DEPENDS ${IO_DEPS}
+      DEPENDS ${IO_DEPS} ${OUT_NET}
```
Comment (litghost) [inline common/cmake/devices.cmake]: https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1011#discussion_r325884470

File: utils/vpr_io_place.py [hunk 2]
```diff
@@ -41,7 +41,7 @@ def constrain_net(self, net_name, loc, comment=""):
             comment=comment,
         )
 
-    def output_io_place(self, f):
+    def output_io_place(self, f, net_to_block=None):
```
Comment (litghost) [inline utils/vpr_io_place.py]: https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1011#discussion_r325885181

**Comment by litghost** [comment]
Looks like you need to add the VPR_DEPS to the IO place after the .net `add_custom_command`: https://sponge.corp.google.com/target?id=6962cf10-82ee-410e-97bc-2b12e47e8252&target=foss-fpga-tools%2Fsymbiflow-arch-defs%2Fpresubmit%2Fxc7&searchFor=

File: utils/vpr_io_place.py (hunk 2 above)
Comment (HackerFoo) [reply utils/vpr_io_place.py]: I moved the code that loads block names from the .net file into `load_block_names_from_net_file` in this class with a docstring.

File: common/cmake/devices.cmake [hunk 3]
```diff
@@ -1101,13 +1138,15 @@ function(ADD_FPGA_TARGET)
...
     append_file_dependency(IO_DEPS ${PINMAP_FILE})
+    append_file_dependency(IO_DEPS ${OUT_NET_REL})
 
     set_target_properties(${NAME} PROPERTIES
         INPUT_IO_FILE ${ADD_FPGA_TARGET_INPUT_IO_FILE})
 
     # Set variables for the string(CONFIGURE) below.
     set(OUT_IO ${OUT_LOCAL}/${TOP}_io.place)
     set(OUT_IO_REL ${OUT_LOCAL_REL}/${TOP}_io.place)
+    set(OUT_NET ${OUT_LOCAL}/${TOP}.net)
```
Comment (litghost) [inline common/cmake/devices.cmake]: This is duplicated, and can be removed?

**Pull request merged**